Backend: Support for API HTTP CORS

techmatt101 opened the pull request:
### Description

Adds server-side CORS support via new flags: --http-cors-allow-origins, --http-cors-allow-methods, and --http-cors-allow-headers. This enables web apps on different domains to call the PhotoPrism API directly, without needing a proxy.

Use case: I self-host PhotoPrism and run a separate photo frame app that fetches images via the PhotoPrism API. Without CORS headers, requests are blocked by the browser.

Note: This is not the same as PHOTOPRISM_CORS_ORIGIN, which controls image access from the PhotoPrism web app to external sources like CDNs.

Please ask me any questions or suggest any code changes as it's first time contribution and writing in go. :) 

[x] Tested with and without the options using RESET Client checking headers return on OPTIONS requests

### Related Issue

- #5133

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/photoprism/photoprism?pullRequest=5132) <br/>All committers have signed the CLA.

techmatt101 committed:
feat: Support for API HTTP CORS

Added HTTP_CORS option to enable OPTION requests to /api endpoints so that it can be called directly in the browser under a different domain without need of a proxy.

techmatt101 committed:
Merge branch 'photoprism:develop' into develop
